std/math.d

```diff
@@ -4103,9 +4103,55 @@ long lrint(real x) @trusted pure nothrow @nogc
 
             return sign ? -result : result;
         }
+        else static if (F.realFormat == RealFormat.ieeeQuadruple)
+        {
+            long result;
+
+            // Rounding limit when casting from real(128-bit) to ulong.
+            enum real OF = 0x1P+112;
+
+            ulong* vl = cast(ulong*)(&x);
+
+            // Find the exponent and sign
+            ulong msb = vl[MANTISSA_MSB];
+            ulong lsb = vl[MANTISSA_LSB];
+            int exp = cast(int)((msb >> 48) & 0x7fff) - 0x3fff;
+            int sign = cast(int)(msb >> 63);
```

Comment:
The `floatTraits` template is supposed to eliminate the need to use "magic numbers" like `0x7fff` in `std.math`'s floating-point code. Admittedly, it feels poorly designed and doesn't really fit everywhere.